src/main/java/dgpt/Ui.java

```diff
@@ -0,0 +1,169 @@
+        for (Task t : taskList.getTaskList()) {
+            System.out.println(index + "." + t.toString());
+            index++;
+        }
+        System.out.println("-----------------------");
+    }
+
+    /**
+     * Displays a message indicating that a task has been marked as done.
+     *
+     * @param t The Task that has been marked as done.
+     */
+    public void showMark(Task t) {
+        System.out.println("-----------------------");
+        System.out.println("DGPT> Nice! I've marked this task as done: ");
+        System.out.println(t.toString());
+        System.out.println("-----------------------");
+    }
+
+    /**
+     * Displays a message indicating that a task has been marked as not done.
+     *
+     * @param t The Task that has been marked as not done.
+     */
+    public void showUnmark(Task t) {
+        System.out.println("-----------------------");
+        System.out.println("DGPT> OK, I've marked this task as not done yet:");
+        System.out.println(t.toString());
+        System.out.println("-----------------------");
+    }
+
+    /**
+     * Displays a message indicating that a task has been deleted from the list.
+     *
+     * @param t The Task that has been removed from the list.
+     * @param size The updated size of the task list after removal.
+     */
+    public void showDelete(Task t, int size) {
+        System.out.println("-----------------------");
+        System.out.println("DGPT> OK, I've removed this task from the list:");
+        System.out.println(t.toString());
+        System.out.printf("Now you have %d tasks in the list.%n", size);
+        System.out.println("-----------------------");
+    }
+
+    /**
+     * Displays a message indicating that a task has been added to the list.
+     *
+     * @param t The Task that has been added to the list.
+     * @param size The updated size of the task list after addition.
+     */
+    public void showTask(Task t, int size) {
+        System.out.println("-----------------------");
+        System.out.println("DGPT> Got it. I've added this task:");
+        System.out.println(t.toString());
+        System.out.printf("Now you have %d tasks in the list.%n", size);
+        System.out.println("-----------------------");
+    }
+
+    /**
+     * Displays a prompt indicating that the user should provide input.
+     */
+    public void showUser() {
+        System.out.print("User> ");
+    }
+
+    /**
+     * Displays a list of tasks that match a search query.
+     *
+     * @param taskList A list of tasks that match the search query.
+     */
+    public void showFind(List<Task> taskList) {
+        System.out.println("-----------------------");
+        System.out.println("Here are the matching tasks in your list:");
+        int index = 1;
+        for (Task t : taskList) {
+            System.out.println(index + "." + t.toString());
+            index++;
+        }
+        System.out.println("-----------------------");
```

Comment:
Maybe consider refactoring this divider print into a helper function? 
Easier to modify in the future if you wish to maybe allow style changes or to adjust to the console width.